### PR TITLE
Opa fmt

### PR DESCRIPTION
**Pull request by mmussomele**
Ready for initial review.

### CODE REVIEW

**Comment by mmussomele** [comment]
Resolves #256

File: format/format.go [hunk 1]
```diff
@@ -0,0 +1,785 @@
...
+
+// Ast formats a Rego AST element. If the passed value is not a valid AST element,
+// Ast returns nil and an error. Ast relies on all AST elements having non-nil
+// Location values, and will return an error if this is not the case.
+func Ast(x interface{}) (formatted []byte, err error) {
+	defer func() {
+		// Ast relies on all terms in the ast element having non-nil Location
+		// values. If a location is nil, Ast will panic, so we need to recover
+		// gracefully.
+		if r := recover(); r != nil {
```
Comment (mmussomele) [inline format/format.go]: we do now

File: ast/parser_ext.go [hunk 2]
```diff
@@ -310,19 +310,26 @@ func ParseStatement(input string) (Statement, error) {
 
 // ParseStatements returns a slice of parsed statements.
 // This is the default return value from the parser.
-func ParseStatements(filename, input string) ([]Statement, error) {
+func ParseStatements(filename, input string) ([]Statement, []*Comment, error) {
 
-	parsed, err := Parse(filename, []byte(input))
+	parsed, err := Parse(filename, []byte(input), GlobalStore(CommentsKey, []*Comment{}))
```
Comment (mmussomele) [inline ast/parser_ext.go]: Actually I'll just add a patch onto this series that does that.

**Comment by tsandall** [comment]
Is this ready to merge?

**Comment by mmussomele** [comment]
Yea, I think so.

**Pull request merged**